### PR TITLE
Harden native MTP launch safety

### PR DESCRIPTION
**Pull request by jjang-ai**
## Summary
- Gate native MTP to lossless greedy text-only requests.
- Carry verifier mode through `DraftStrategy.nativeMTP` and server runtime resolution.
- Add hybrid safety warmup / AR fallback telemetry so hard prompts stay byte-identical instead of using an unsafe fast path.

## Local proof
- `swift build -c release --product RunBench --jobs 2` passed.
- Qwen3.6-35B-A3B-MXFP4-MTP hard prompts, cache coordinator + hybrid + disk/SSM rederive: p1/p2/p3 were byte-identical to AR, no loops, no reasoning leaks. Artifact: `docs/local/mtp-debug-run/20260519T152646-hardcheck-current`.
- Count sanity: MTP d3 activated and stayed byte-identical, AR 91.7 tok/s vs MTP 115.2 tok/s, `avgCommittedPerVerify=3.94`, artifact `docs/local/mtp-debug-run/20260519T153800-count-current`.

## Notes
This intentionally prioritizes correctness over forcing the old unsafe chunk path. Hard prompts fall back/warm up and can be slower; predictable prompts still show d3 speedup.

### CODE REVIEW

**Comment by jjang-ai** [comment]
Crash follow-up evidence for this branch:

- osaurus 0.18.35 crash report `/Users/eric/Downloads/crash.txt` symbolicates thread 25 into `NativeMTPTokenIterator.batchedGreedyTargetTokenIds(logits:count:)` via `MLXArray.asArray` / `mlx_array_data_uint8`.
- This is native MTP verifier materialization, not generic load or server-panel code. The crash text does not itself name MXFP8; the failing installed app path was using the native-MTP decode verifier path.
- Added a short-logits regression for the greedy verifier and fallback to sequential verifier when the batched materialization path cannot prove the expected shape/count.

Local verification after commit `da815b8`:

```
DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer xcodebuild build -scheme MLXLMCommon -destination platform=macOS,arch=arm64 -derivedDataPath /tmp/vmlx-native-mtp-build
# ** BUILD SUCCEEDED **

DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer swift test --filter MTPRuntimeFocusedTests/nativeMTPGreedyVerifierRejectsShortLogitsBeforeMaterialization
# Swift Testing: 1 test passed

BENCH_PERF=1 BENCH_PERF_WARMUP=0 BENCH_PERF_RUNS=1 BENCH_MODEL=/Users/eric/models/JANGQ/Qwen3.6-27B-MXFP8-MTP BENCH_PERF_PATH=iter BENCH_PERF_NATIVE_MTP_DEPTH=3 BENCH_MAX_TOKENS=16 BENCH_PERF_PROMPT='Reply with exactly: MTP-SMOKE-OK' BENCH_PERF_FULL_TEXT=1 .build/debug/RunBench
# TEXT_PREVIEW "MTP-SMOKE-OK"
# genTokens=7 genSec=33.641 tokps=0.2 peak_footprint_mib=1519 loop=NO leaks=none
```

Note: the smoke proves crash-safety and coherent output for this case, not a production-speed native-MTP row. It is still too slow at 0.2 tok/s.

**Comment by jjang-ai** [comment]
Follow-up pushed in `1f883dd`:

- Adds `ModelRuntimeDetectionSnapshot`, a no-load trace for bundle classification inputs: model type, text_config fallback, JANG/JANGTQ/MXFP weight format signals, mxtq bits source, sidecar presence, preprocessor/VL signals, and native-MTP status.
- Wires the trace into `ModelRuntimeCapabilitySnapshot` for local/resolved configurations without loading model weights.
- Adds focused synthetic coverage proving JANGTQ+VL+native-MTP classification and MXFP-vs-affine-JANG separation.

Verification:
- `DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer swift test --filter VMLXUmbrellaProductTests`
- `DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer xcodebuild build -scheme MLXLMCommon -destination 'platform=macOS,arch=arm64' -derivedDataPath /tmp/vmlx-runtime-detection-build`

**Comment by jjang-ai** [comment]
Additional autodetect fix pushed in `0deffe2`:

- Local inventory found `/Users/eric/models/dealign.ai/Qwen3.6-27B-MXFP4-CRACK` has no `weight_format`, but does stamp `config.quantization.mode=mxfp4`.
- Detection now classifies MXFP from `quantization.mode` / `text_config.quantization.mode` as well as `weight_format`, and records those fields in the trace evidence.
- The MXFP-vs-affine-JANG test now covers this exact missing-weight-format shape.

Verification repeated:
- `DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer swift test --filter VMLXUmbrellaProductTests`
- `DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer xcodebuild build -scheme MLXLMCommon -destination 'platform=macOS,arch=arm64' -derivedDataPath /tmp/vmlx-runtime-detection-build`
